data/api/client-server/content-repo.yaml

```
@@ -12,154 +12,164 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-swagger: '2.0'
+openapi: 3.1.0
 info:
-  title: "Matrix Client-Server Content Repository API"
-  version: "1.0.0"
-host: localhost:8008
-schemes:
-  - https
-  - http
-basePath: /_matrix/media/v3
-consumes:
-  - application/json
-  - "*/*"
-produces:
-  - application/json
-  - "*/*"
-securityDefinitions:
-  $ref: definitions/security.yaml
+  title: Matrix Client-Server Content Repository API
+  version: 1.0.0
 paths:
-  "/upload":
+  /upload:
     post:
       summary: Upload some content to the content repository.
       operationId: uploadContent
-      produces: ["application/json"]
       security:
         - accessToken: []
       parameters:
         - in: header
           name: Content-Type
-          type: string
           description: The content type of the file being uploaded
-          x-example: "application/pdf"
+          example: application/pdf
+          schema:
+            type: string
         - in: query
-          type: string
-          x-example: "War and Peace.pdf"
           name: filename
           description: The name of the file being uploaded
-        - in: body
-          name: "<content>"
-          description: The content to be uploaded.
-          required: true
-          x-example: "<bytes>" # so the spec shows "<bytes>" without quotes.
+          example: War and Peace.pdf
           schema:
             type: string
-            example: "<bytes>"
-            format: byte
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: string
+              example: <bytes>
+              format: byte
+          "*/*":
+            schema:
+              type: string
+              example: <bytes>
+              format: byte
```

AIUI, the OpenAPI v3 spec [recommends](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.3.md#data-types) using `format: byte` for base64-encoded payloads and `format: binary` for "any sequence of octets". The latter is more appropriate across content-repo.
```suggestion
              format: binary
```

---

Same as above.